pkgs/applications/audio/youtube-music/default.nix

```diff
@@ -24,54 +21,17 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-nxpctEG4XoxW6jOAxGdgTEYr6YnhFRR8+5HUQLxRJB0=";
   };
 
-  pnpmDeps = stdenvNoCC.mkDerivation {
-    pname = "${finalAttrs.pname}-pnpm-deps";
-    inherit (finalAttrs) src version ELECTRON_SKIP_BINARY_DOWNLOAD;
-
-    nativeBuildInputs = [ jq moreutils nodePackages.pnpm cacert ];
-
-    installPhase = ''
-      export HOME=$(mktemp -d)
-
-      pnpm config set store-dir $out
-      pnpm install --frozen-lockfile --ignore-script
-
-      rm -rf $out/v3/tmp
-      for f in $(find $out -name "*.json"); do
-        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
-        jq --sort-keys . $f | sponge $f
-      done
-    '';
-
-    dontBuild = true;
-    dontFixup = true;
-    outputHashMode = "recursive";
-    outputHash = {
-      x86_64-linux = "sha256-bujlQxP6Lr3qPUDxYXKyb702ZJY/xbuCsu3wVDhcb+8=";
-      aarch64-linux = "sha256-0kyjjttpXpFVhdza5NAjGrRn++qc/N5/u2dQl7VufLE=";
-      x86_64-darwin = "sha256-Q37QJt/mhfpSguOlkJGKFTCrIOrpbG3OBwaD/Bg09Us=";
-      aarch64-darwin = "sha256-wbfjzoGa/6vIlOOVX3bKNQ2uxzph3WSofo3MGXqA6yQ=";
-    }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
```

Comment:
We should check that there is no hash mismatch for different systems, now when `fetchPnpmDeps` is used.

Reply:
Ofbrorg is reporting the hashes mismatch:

On aarch64-linux:

```
error: hash mismatch in fixed-output derivation '/nix/store/qq4ncfffg02xdx2laqcx1fclk0vjqxfn-youtube-music-pnpm-6_0-deps.drv':
         specified: sha256-EMIzOzX1QDNDcGRTV189SXiH+lZp0PgAFX5IERBuilI=
            got:    sha256-egfteCl9ThMNz/mlYVFQTfwmIwHIbP1/0V2dhV8BL0E=
```

Later ofborg will report the hash mismatches from other platforms.

Reply:
I see. Interestingly enough this does not affect all packages here. Vesktop for example works fine when built on Linux aarch64 and x86_64.

The issue seems to be a "side effects cache" where some packages produce platform-dependent outputs. It seems like I can disable that though, which might work here

Reply:
Could both of you try again?

Reply:
1 package built:
youtube-music

Result of `nixpkgs-review pr 290715` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>youtube-music</li>
  </ul>
</details>

Reply:
> Could both of you try again?

CI seems totally green for `{aarch64,x86_64}-linux`, but there is a wierd error on `x86_64-darwin`.

Reply:
That error is #294725, we can ignore it for now.